tests/rptest/tests/basic_kafka_compat_tests.py

```diff
@@ -0,0 +1,149 @@
+            f"Redpanda result: {rp_result}, kafka result: {kafka_result}")
+        assert rp_result == kafka_result, "Kafka and Redpanda results differs"
+
+
+class TxKafkaCompatTest(KafkaCompatTest):
+    @cluster(num_nodes=7)
+    @parametrize(metadata_quorum='COLOCATED_KRAFT')
+    def test_concurrent_producer_ids(self, metadata_quorum):
+        def test_case(broker_service: RedpandaService | KafkaServiceAdapter):
+
+            partition_count = 1
+            topic = TopicSpec(name="test-topic",
+                              replication_factor=3,
+                              partition_count=partition_count)
+
+            DefaultClient(broker_service).create_topic(topic)
+
+            producer_1 = ck.Producer(
+                {
+                    'bootstrap.servers': broker_service.brokers(),
+                    'transactional.id': "tx-id-1",
+                    'client.id': "prod-1"
+                },
+                logger=self.logger,
+                debug='all')
+
+            self.logger.info("Initializing transactions for producer-1")
+            producer_1.init_transactions()
+            producer_1.begin_transaction()
+            for i in range(5):
+                self.logger.info(f"Producing message {i} from producer-1")
+                producer_1.produce(topic.name, key="k", value=f"v-{i}")
+
+            self.logger.info("Producer-1 committing transaction...")
+            producer_1.commit_transaction()
+
+            self.logger.info("Initializing transactions for producer-2")
+            producer_2 = ck.Producer(
+                {
+                    'bootstrap.servers': broker_service.brokers(),
+                    'transactional.id': "tx-id-1",
+                    'client.id': "prod-2"
+                },
+                logger=self.logger,
+                debug='all')
+            producer_2.init_transactions()
+
+            producer_2.begin_transaction()
+            for i in range(5):
+                producer_2.produce(topic.name, key="k", value=f"v-{i}")
+
+            producer_2.commit_transaction()
+
+            consumer = ck.Consumer(
+                {
+                    "bootstrap.servers": broker_service.brokers(),
+                    "group.id": "group-1",
+                    'auto.offset.reset': 'earliest',
+                    'enable.auto.commit': True,
+                    'isolation.level': 'read_committed',
+                },
+                logger=self.logger,
+            )
+            consumer.subscribe([topic.name])
+
+            msgs = []
+            timeout = time.time() + 30
+            while len(msgs) < 9:
+                msg = consumer.poll(timeout=1)
+                if time.time() > timeout:
+                    break
+
+                if msg:
+                    self.logger.info(
+                        f"consumed: {msg.value()} at offset: {msg.offset()}")
+                    msgs.append((msg.value(), msg.offset()))
+            consumer.close()
```

Comment:
In this case the callable is not returning anything, so nothing matches? Intentional?

Also I don't see this running in CI 🤔